src/drv_nxtvr/nxtvr.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * OpenHMD - Free and Open Source API and drivers for immersive technology.
+ * Copyright (C) 2019 Vis3r.
+ * Distributed under the Boost 1.0 licence, see LICENSE for full text.
+ */
+
+/*  Nxtvr Driver */
+
+//Using as code base: https://github.com/der-b/OpenHMD/tree/master/src/drv_sparkfun_9dof
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <hidapi/hidapi.h>
+#include <openhmdi.h>
+
+#include <omath.h>
+
+#include "nxtvr.h"
+
+#define FEATURE_BUFFER_SIZE 256
+
+#define HID_REPORT_QUAT_ID 42
+#define HID_REPORT_QUAT_SIZE 17
+
+// These ID's have to be used only for debugging purposes
+#define VENDOR_ID 0x1eaf  // Default Arduion_STM32 VID
+#define PRODUCT_ID 0x0024 // Default Arduion_STM32 PID
+
+#define DEVICE_NAME "NxtVR"
+
+#undef LOGLEVEL
+#define LOGLEVEL 0
+
+typedef struct
+{
+    ohmd_device base;
+    hid_device *handle;
+} nxt_priv;
+
+static void handle_nxtvr_hmd_sensor_msg(nxt_priv *priv, unsigned char *buffer, int size)
+{
+    if (size < 8)
+    {
+        LOGE("NXTVR: REPORT size has to be at least 8!");
+        return;
+    };
+
+    uint8_t reportID = (uint8_t)buffer[0];
+    if (reportID == HID_REPORT_QUAT_ID)
+    {
+        if (size != HID_REPORT_QUAT_SIZE)
+        {
+            LOGE("NXTVR: Invalid data size %d for reportID 42.", size);
+            return;
+        };
+
+        quatf quater;
+        //copy remaining 16 bytes into float union
+        memcpy((void *)&quater, (void *)&buffer[1], 16);
+
+        priv->base.rotation = quater;
```

Comment:
This seems to use pre-calculated quaternions, where OpenHMD prefers to use raw accel and gyro data from the IMU to be able to use our own Sensor Fusion and allow for more and easier optimization.

Also using memcpy to get data from your buffer into the driver quaternion is bad practice.
A preferred method is to use the read functions that we implemented in most drivers.
https://github.com/OpenHMD/OpenHMD/blob/master/src/drv_nolo/packet.c#L97